### ISSUE TITLE
Feature request - allow for "ignorescope" in rules

### ISSUE THREAD

**Issue by aireilly** [jdkato/txtlint#561]
We have a new rule that [checks for pascal and camel case](https://github.com/redhat-documentation/vale-at-red-hat/blob/main/.vale/styles/RedHat/PascalCamelCase.yml) and suggests marking up instances with backticks. It would be nice to be able to ignore certain scopes, for example headings. 

Example rule with suggested "ignore scope":

```yaml
---
extends: existence
scope: text
ignorescope: heading
ignorecase: false
level: suggestion
link: https://redhat-documentation.github.io/asciidoc-markup-conventions
message: "Consider wrapping this Pascal or Camel case term ('%s') in backticks."
source: https://github.com/redhat-documentation/vale-at-red-hat/tree/main/.vale/styles/RedHat/PascalCamelCase.yml
```

**Comment by jdkato** [comment]
This is now possible. 

An updated docs section on scoping is in the works, but the basic idea is that any scope prefaced with "~" is _negated_:

```yaml
message: "'%s' left in text"
extends: existence
scope:
  # all scopes != h2
  - ~heading.h2
level: error
tokens:
  - TODO
```
You can combine this normal  (non-negated) scoping to match any selection of scopes you want.

**Comment by aireilly** [comment]
Wow, excellent. Thanks for the quick turn around!

**Comment by aireilly** [comment]
So something like this work should work, for example? 

```yaml
scope: 
  - list
  - sentence
  - ~heading.h2
```
